x-pack/filebeat/input/awss3/metrics.go

```diff
@@ -40,16 +45,24 @@ func (m *inputMetrics) Close() {
 	m.unregister()
 }
 
+func (m *inputMetrics) Initialize() {
```

Comment:
I recommend to create a method specifically for setting the "sqs messages waiting" value that encapsulates the logic of initializing the value at the time of first use. Then change any code to avoid direct access to `sqsMessagesWaiting`. Like

```
func (m *inputMetrics) setSQSMessagesWaiting(count int64) {
    m.initSQSMessageWaitingOnce.Do(func() {
        m.sqsMessagesWaiting = monitoring.NewInt(m.registry, "sqs_messages_waiting_gauge")
    })

    m.sqsMessagesWaiting(count)
}
```